clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/caching/agents/TitusV2ClusterCachingAgent.java

```diff
@@ -0,0 +1,634 @@
+   * of an increase in time before a change becomes visible in the UI.
+   *
+   * A change will not be visible until a caching cycle has completed.
+   */
+  private OnDemandResult minimalOnDemand(ProviderCache providerCache, Job job, Map<String, ?> data) {
+    String serverGroupKey = Keys.getServerGroupV2Key(job.getName(), account.getName(), region.getName());
+    Map<String, Collection<CacheData>> cacheResults = new HashMap<>();
+    if (job == null) {
+      // avoid writing an empty onDemand cache record (instead delete any that may have previously existed)
+      providerCache.evictDeletedItems(ON_DEMAND.ns, Collections.singletonList(serverGroupKey));
+    } else {
+      Map <String, Object> attributes = new HashMap<>();
+      attributes.put("cacheTime", new Date());
+      attributes.put("cacheResults", Collections.emptyMap());
+
+      CacheData cacheData = metricsSupport.onDemandStore( () -> new DefaultCacheData(
+        serverGroupKey,
+        10 * 60, // ttl is 10 minutes
+        attributes,
+        Collections.emptyMap()
+      ));
+
+      cacheResults.put(ON_DEMAND.ns, Collections.singletonList(cacheData));
+    }
+    Map<String, Collection<String>> evictions = job == null
+      ? Collections.emptyMap()
+      : Collections.singletonMap(SERVER_GROUPS.ns, Collections.singletonList(serverGroupKey));
+
+    log.info("minimal onDemand cache refresh (data: {}, evictions: {})", data, evictions);
+    return new OnDemandResult(getOnDemandAgentType(), new DefaultCacheResult(cacheResults), evictions);
+  }
+
+  @Override
+  public CacheResult loadData(ProviderCache providerCache) {
+    Long startTime = System.currentTimeMillis();
+
+    log.info("Describing items in {}", getAgentType());
+    List<CacheData> evictFromOnDemand = new ArrayList<>();
+    List<CacheData> keepInOnDemand = new ArrayList<>();
+
+
+    List<Job> jobs = titusClient.getAllJobsWithoutTasks();
+    PercentileTimer
+      .get(registry, metricId.withTag("operation", "getAllTasks"))
+      .record(System.currentTimeMillis() - startTime, MILLISECONDS);
+
+    Long startScalingPolicyTime = System.currentTimeMillis();
+    List<ScalingPolicyResult> scalingPolicyResults = titusAutoscalingClient != null
+      ? titusAutoscalingClient.getAllScalingPolicies()
+      : Collections.emptyList();
+    PercentileTimer
+      .get(registry, metricId.withTag("operation", "getScalingPolicies"))
+      .record(System.currentTimeMillis() - startScalingPolicyTime, MILLISECONDS);
+
+    Long startLoadBalancerTime = System.currentTimeMillis();
+    Map<String, List<String>> allLoadBalancers = titusLoadBalancerClient.getAllLoadBalancers();
+    PercentileTimer
+      .get(registry, metricId.withTag("operation", "getLoadBalancers"))
+      .record(System.currentTimeMillis() - startLoadBalancerTime, MILLISECONDS);
+
+    Long startJobIdsTime = System.currentTimeMillis();
+    Map<String, List<String>> taskAndJobIds = titusClient.getTaskIdsForJobIds();
```

Comment:
Since Titus tasks don't know their job ids, we need to grab them separately from Titus to build relationships. This call takes < 1s running locally against Titus us-east-1